wp-dependency-installer.php

```diff
@@ -147,12 +147,12 @@ public function register( $config, $plugin_path ) {
 			$source = basename( $plugin_path );
 			foreach ( $config as $dependency ) {
 				$dependency['source']    = $source;
+				$dependency['sources'][] = $source;
 				$slug                    = $dependency['slug'];
-				$dependency['tooltip'][] = $source;
+				if ( isset( $this->config[ $slug ] ) && $this->is_required( $dependency ) ) {
+					$dependency['sources'] = array_merge( $this->config[ $slug ]['sources'], $dependency['sources'] );
+				}
```

Comment:
You beat me to this. I was thinking about it after I finished for the night. 😉

Reply:
However I'm not really sure that it works properly ( this thing https://github.com/afragen/wp-dependency-installer/issues/53 is making me crazy )

Reply:
You keep adding good things to this PR and I’m still trying to convince myself that it’s not simply too much added complexity.

Reply:
> However I'm not really sure that it works properly ( this thing https://github.com/afragen/wp-dependency-installer/issues/53 is making me crazy )

There should be no other problems with current solution.

> You keep adding good things to this PR and I’m still trying to convince myself that it’s not simply too much added complexity.

Some changes are here only to encourage you to have a little more reusable code..